internal/resources/google/container_cluster.go

```diff
@@ -0,0 +1,200 @@
+		costComponents = append(costComponents, r.autopilotStorageCostComponent())
+	}
+
+	subresources := []*schema.Resource{}
+
+	if r.DefaultNodePool != nil {
+		s := ContainerNodePool{
+			Address:      r.DefaultNodePool.Address,
+			Region:       r.DefaultNodePool.Region,
+			Zones:        r.DefaultNodePool.Zones,
+			CountPerZone: r.DefaultNodePool.CountPerZone,
+			NodeConfig:   r.DefaultNodePool.NodeConfig,
+		}
+		s.PopulateUsage(r.DefaultNodePool.UsageData)
+
+		subresources = append(subresources, s.BuildResource())
+	}
+
+	for _, nodePool := range r.NodePools {
+		s := ContainerNodePool{
+			Address:      nodePool.Address,
+			Region:       nodePool.Region,
+			Zones:        nodePool.Zones,
+			CountPerZone: nodePool.CountPerZone,
+			NodeConfig:   nodePool.NodeConfig,
+		}
+		s.PopulateUsage(nodePool.UsageData)
+
+		subresources = append(subresources, s.BuildResource())
+	}
+
+	return &schema.Resource{
+		Name:           r.Address,
+		UsageSchema:    ContainerClusterUsageSchema,
+		CostComponents: costComponents,
+		SubResources:   subresources,
+	}
+}
+
+// managementFeeCostComponent returns a cost component for cluster management
+// fee.
+func (r *ContainerCluster) managementFeeCostComponent() *schema.CostComponent {
+	description := "Regional Kubernetes Clusters"
+	name := "Cluster management fee"
+
+	if r.IsZone {
+		description = "Zonal Kubernetes Clusters"
+	}
+
+	if r.AutopilotEnabled {
+		description = "Autopilot Kubernetes Clusters"
+		name = "Autopilot"
+	}
+
+	return &schema.CostComponent{
+		Name:           name,
+		Unit:           "hours",
+		UnitMultiplier: decimal.NewFromInt(1),
+		HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("gcp"),
+			Region:        strPtr("global"),
+			Service:       strPtr("Kubernetes Engine"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", Value: strPtr(description)},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			StartUsageAmount: strPtr("0"),
+			EndUsageAmount:   strPtr(""),
+		},
+	}
+}
+
+// autopilotCPUCostComponent returns a cost component for Autopilot vCPU usage.
+func (r *ContainerCluster) autopilotCPUCostComponent() *schema.CostComponent {
+	var quantity *decimal.Decimal
+
+	if r.AutopilotVCPUCount != nil {
+		quantity = decimalPtr(decimal.NewFromFloat(*r.AutopilotVCPUCount))
+	}
+
+	return &schema.CostComponent{
+		Name:           "Autopilot vCPU",
+		Unit:           "vCPU",
+		UnitMultiplier: schema.HourToMonthUnitMultiplier,
+		HourlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("gcp"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("Kubernetes Engine"),
+			ProductFamily: strPtr("Compute"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", ValueRegex: regexPtr("^Autopilot Pod mCPU Requests")},
```

Comment:
Pricing of CPAPI differs from Google Pricing page by x1000: for us-central1 region CPAPI has `0.000044500 $/hour` (0.032485 $/mo), the pricing page/calculator show `0.0445 $/hour` (32.49 $/mo).

https://cloud.google.com/kubernetes-engine/pricing